### PR TITLE
Leaderboard

### PR DESCRIPTION
**Pull request by MilStancheva**
Home page leaderboard is added. Clubs are filtering by school-year and are ordered by their points. The points are dynamically calculated depending on the missions for the concrete school year the club has organized and the number of the heroes participating in the mission. 

The club details can be seen when clicking on a club on the leaderboard. Styles are not finished yet as we are waiting for the final  design.

### CODE REVIEW

File: HeroesCup.Modules/ClubsModule/Services/SchoolYearService.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+namespace ClubsModule.Services
+{
+    public class SchoolYearService : ISchoolYearService
+    {
+        public string GetCurrentSchoolYear()
+        {
+            return CalculateSchoolYear(DateTime.UtcNow);
+        }
+
+        public string CalculateSchoolYear(DateTime startDate)
```
Comment (adautev) [inline HeroesCup.Modules/ClubsModule/Services/SchoolYearService.cs:13]: Add a unit test for this one.

File: HeroesCup.Web/Services/LeaderboardService.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            Regex initials = new Regex(@"(\b[a-zA-Z-а-яА-Я])[a-zA-Z-а-яА-Я]* ?");
+            return initials.Replace(organizationName, "$1");
+        }
+
+        private int GetHeroesCount(Club club)
+        {
+            return club.Heroes.Count;
+        }
+
+        private int getClubPoints(IEnumerable<Mission> missions)
```
Comment (adautev) [inline HeroesCup.Web/Services/LeaderboardService.cs:116]: Either open up and test or test method that uses this one. (:

File: HeroesCup.Modules/ClubsModule/Services/SchoolYearService.cs (hunk 1 above)
Comment (MilStancheva) [reply HeroesCup.Modules/ClubsModule/Services/SchoolYearService.cs:13]: Basic unit tests are added.

File: HeroesCup.Web/Services/LeaderboardService.cs (hunk 2 above)
Comment (MilStancheva) [reply HeroesCup.Web/Services/LeaderboardService.cs:123]: Basic unit tests for the method that uses the getClubPoints (GetClubsBySchoolYearAsync) are added.

**Pull request merged**